dirs/dirs.go

```diff
@@ -179,7 +183,8 @@ func SetRootDir(rootdir string) {
 		DistroLibExecDir = filepath.Join(rootdir, "/usr/lib/snapd")
 	}
 
-	CoreLibExecDir = filepath.Join(rootdir, "/usr/lib/snapd")
+	CoreLibExecDir = "/usr/lib/snapd"
+	CoreSnapMountDir = "/snap"
```

Comment:
We need a variable which has just the clean mount dir without the root prefix as otherwise we end up with something $prefix/$prefix/$snapmountdir in the unit tests. See https://github.com/snapcore/snapd/pull/3222/files#diff-76f691a17e89c1330742e51bb468ef76R203 as an example. The only place where is ever being used are unit tests. I can move these variables into test code and keep it away from dirs/dirs.go actually would be nice to keep these things in a central place.

Also CoreLibExecDir are needed as these describe where in the core snap those directories are located. These are different on Fedora than the outside SnapMountDir/LibExecDir so we have to keep them distinct.